### DISCUSSION TITLE
Merge (Replace) one array to (by) other with dynamic size

### DISCUSSION THREAD

**Discussion by gsaha009** [scikit-hep/awkward#3320] [General]
Hello Experts,

I was trying to do an operation as described in the following:

I have two arrays
```
a = [ [0], [1], [1], [0] ]
b = [ [12.4], [23.5] ]
```
Now, I want to merge `b` to `a` where the value `1` matches. I mean I want to have something like this:
`c = [ [], [12.4], [23.5], [] ]`

The reason is the following:
I had some awkward arrays in the beginning. In order to do some existing operations, I had to convert it to numpy and, thats why I applied a mask to make that array regular. But, I saved the places where the new elements should be placed. 

Please let me know if there is any way to do the same.

Thank you very much

Gourab

**Comment by agoose77** [comment]
If I understand your question, you have an array `a` that has been made regular (fixed-dimension). You have another array `b` that represents an "original" array whose contents need to be inserted _back_ into `a`? Is that correct? i.e., is the number of `1` in your array `a` equal to the number of elements in array `b`?


It would be good to understand more about your problem; the shapes and dimensions of the arrays are important. But, we can make a start!
I only have a few minutes here, so I'm sure that @jpivarski or @pfackeldey will expand / provide alternatives.

What it _seems_ like you want to do is to build an _index_ that can pull out the proper elements of `b` with the right structure:
```python
import awkward as ak
import numpy as np

flat_a = ak.flatten(a)
shape_a = ak.num(a)

flat_b = ak.flatten(b)

masked_flat_a = flat_a.mask[flat_a == 1]
flat_index = np.cumsum(masked_flat_a) - 1

flat_result = flat_b[flat_index]
result = ak.unflatten(flat_result, shape_a)
result
```

This code sample flattens both arrays (preserving the shape of `a` in `shape_a`), masks out the "valid" elements of `a` (so that the resulting array has `None` for the elements that are "padding" (0)), builds a zero-based index into `b` by counting each non-zero element of `masked_flat_a`, and then indexes into `flat_b`. Inspecting `flat_index` should show you how it works!

This solution uses Awkward's ragged indexing feature, which is very powerful.
